File: JAVA/Day12.java
class Grade extends Student{
   private int score;
    
   Grade(String fname,String lname,int p,int s){
         super(fname, lname, p);
         score=s;
   }
    
   public char calculate(){
       if(score<40){
           return 'D';
       }else if(score>=40 && score<60){
           return 'B';
       }else if(score>=60 && score<75){
           return 'A';
       }else if(score>=75 && score<90){
           return 'E';
       }else if(score>=90 && score<=100){
           return 'O';
       }
       return 'E';
   }
   
}

File: RUBY/day12.rb
class Grade < Student
  attr_accessor :calculate
  @@grades = [
      [ 1..39,   'D' ],
      [ 40..59,  'B' ],
      [ 60..74,  'A' ],
      [ 75..89,  'E' ],
      [ 90..100, 'O' ],
  ]
  
  def initialize(firstName, lastName, phone, score)
    super(firstName, lastName, phone)
    @calculate = @@grades.each.reduce(nil) do |grade, grades|
      grade || (grades[0] === score && grades[1])
    end
    raise ArgumentError "invalid score: #{score}" if @calculate.nil?
  end
end
